core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -204,7 +204,11 @@ private TableMetadata internalApply() {
     base.snapshots().stream()
         .map(Snapshot::snapshotId)
         .filter(snapshot -> !idsToRetain.contains(snapshot))
-        .forEach(idsToRemove::add);
+        .forEach(
+            snapshotId -> {
+              updatedMetaBuilder.removeStatistics(snapshotId);
```

Comment:
This should be handled inside of `removeSnapshots` because we `TableMetadata` that to maintain consistency.

Reply:
Ah, true. I will move it.